quinn/src/connection.rs

```diff
@@ -400,19 +404,35 @@ impl Connection {
 
     /// Close the connection immediately.
     ///
-    /// Pending operations will fail immediately with [`ConnectionError::LocallyClosed`]. Delivery
-    /// of data on unfinished streams is not guaranteed, so the application must call this only
-    /// when all important communications have been completed, e.g. by calling [`finish`] on
-    /// outstanding [`SendStream`]s and waiting for the resulting futures to complete.
+    /// Pending operations will fail immediately with [`ConnectionError::LocallyClosed`]. No
+    /// more data is sent to the peer and the peer may drop any stream data upon receiving
```

Comment:
```suggestion
    /// more data is sent to the peer and the peer may drop buffered data upon receiving
```
Streams aren't the only form of application data, and we should be clear that we're talking about data the peer already received at the transport layer, i.e. buffered data.